### ISSUE TITLE
Trim unused assemblies

### ISSUE THREAD

**Issue by PlayCreatively** [MonoGame/MonoGame#7560]
I'm trying to publish with "**Trim unused assemblies**" on. I get this error and I'm hoping someone could intuit what this could be.

> 2>Fatal error in IL Linker
> 2>Unhandled exception. Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'Microsoft.Xna.Framework.Content.ContentTypeReader, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
> 2> ---> System.IO.FileNotFoundException: Unable to find 'Microsoft.Xna.Framework.Content.ContentTypeReader.dll' or 'Microsoft.Xna.Framework.Content.ContentTypeReader.exe' file

#### What version of MonoGame does the bug occur on:
- MonoGame 3.8.0.1641

#### What operating system are you using:
- Windows 10

#### What MonoGame platform are you using:
- DesktopGL

**Comment by mrhelmut** [comment]
Hey there, this is due to the trimmer removing assemblies which are loaded through reflection, typically type readers. You need to exclude MonoGame from being trimmed, please check the [.NET documentation](https://docs.microsoft.com/en-us/dotnet/core/deploying/trim-self-contained).

**Comment by PlayCreatively** [comment]
Interestingly enough, this snipped was hiding in my project file
```
  <ItemGroup>
    <TrimmerRootAssembly Include="Microsoft.Xna.Framework.Content.ContentTypeReader" Visible="false" />
  </ItemGroup>
```
which is why I got this error

> 'Microsoft.Xna.Framework.Content.ContentTypeReader, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
> 2> ---> System.IO.FileNotFoundException: Unable to find 'Microsoft.Xna.Framework.Content.ContentTypeReader.dll' or 'Microsoft.Xna.Framework.Content.ContentTypeReader.exe' file

So Microsoft's way of [excluding assemblies from being trimmed](https://docs.microsoft.com/en-us/dotnet/core/deploying/trim-self-contained#prevent-assemblies-from-being-trimmed) is the reason for this error.
Exactly why this causes an error, I'm not sure.

Thank you for the response :)

**Comment by mrhelmut** [comment]
This has been fixed in #7529. Please replace ```Microsoft.Xna.Framework.Content.ContentTypeReader``` with ```MonoGame.Framework```.
